eq_cir_proxy_service/routers/instrument_router.py

```diff
@@ -0,0 +1,47 @@
+"""Module defines the instrument router for handling requests related to instruments in the EQ CIR Proxy Service."""
+
+from uuid import UUID
+
+from fastapi import APIRouter, HTTPException, Path, Query
+
+from eq_cir_proxy_service.config.logging_config import logging
+from eq_cir_proxy_service.exception import exception_messages
+from eq_cir_proxy_service.services.schema import schema_retrieval_service
+from eq_cir_proxy_service.services.validators.request_validator import (
+    validate_version,
+)
+from eq_cir_proxy_service.types.custom_types import Schema
+
+router = APIRouter()
+logger = logging.getLogger(__name__)
+INSTRUMENT_ID_PATH = Path(..., description="UUIDv4 of the instrument")
+
+
+@router.get("/instrument/{instrument_id}")
+async def get_instrument_by_uuid(
+    instrument_id: UUID = INSTRUMENT_ID_PATH,
+    version: str = Query(default=None, description="Optional version of the instrument"),
+) -> Schema:
+    """Retrieve an instrument by its UUID and optional version."""
+    logger.info("Receiving the instrument id...")
+    logger.debug("Received instrument id: %s", instrument_id)
+
+    try:
+        if version is not None:
```

Comment:
What if it's something else then the `if` statement condition? Then the try will be successfully completed? Should we try without the version? Should version existance be checked earlier?

Reply:
Not sure I understand.
If the version is None, ie not provided, it skips straight to the return calling schema_retrieval_service, where version is passed as None.
If the version is not None, it validates the version, and if the validation doesn't throw an error it moves on to the return, passing the validate version to the schema_retrieval_service call